### PR TITLE
[MO] - Building Images

### PR DESCRIPTION
**Pull request by see-quick**
This PR adding support for building, and publishing images to the external registry. As we decided on the community call we are going to have multiple support of Kafka versions with fixed Scala version (latest).  

Images examples:
```
REPOSITORY                           TAG                    IMAGE ID           CREATED             SIZE
strimzi/test-container-images          0.0.1-kafka-2.7.0      e2fbd2918cfa   18 minutes ago      444MB
strimzi/test-container-images          0.0.1-kafka-2.8.0      9dac3b61e03e   18 minutes ago      448MB
```

### CODE REVIEW

File: supported_scala.version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.12
```
Comment (scholzj) [inline supported_scala.version:1]: 2.13?

File: images/build_images.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+echo "Building images with DOCKER_VERSION_ARG=$DOCKER_VERSION_ARG, PROJECT_NAME=$PROJECT_NAME, DOCKER_TAG=$DOCKER_TAG, DOCKERFILE_DIR=$DOCKERFILE_DIR"
+
+#####
+# FOR EACH KAFKA VERSION DOWNLOAD BINARIES
+#####
+for KAFKA_VERSION in $KAFKA_VERSIONS
+do
+    for SCALA_VERSION in $SCALA_VERSIONS
+    do
```
Comment (scholzj) [inline images/build_images.sh:30]: This does not work ... you cannot loop through Scala versions if your docker tag doesn't include Scala version as well.

File: images/download_kafka.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+KAFKA_VERSIONS=$(cat supported_kafka.version)
+SCALA_VERSIONS=$(cat supported_scala.version)
+
+#####
+# FOR EACH KAFKA VERSION DOWNLOAD BINARIES
+#####
+for KAFKA_VERSION in $KAFKA_VERSIONS
+do
+    for SCALA_VERSION in $SCALA_VERSIONS
+    do
```
Comment (scholzj) [inline images/download_kafka.sh:19]: Same as above, this does not work with the current tag.

File: images/download_kafka.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        echo "Downloading: "$KAFKA_VERSION" with $SCALA_VERSION."
+        KAFKA_URL="https://dlcdn.apache.org/kafka/$KAFKA_VERSION/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz"
+        mkdir -p $KAFKA_DOWNLOADED_TARS_DIR && wget $KAFKA_URL -P "$KAFKA_DOWNLOADED_TARS_DIR"
+        echo "kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz"
+        echo "Extracting: "$KAFKA_VERSION" with $SCALA_VERSION."
+        mkdir -p $KAFKA_BINARIES_DIR && tar -xvzf $KAFKA_DOWNLOADED_TARS_DIR/kafka_$SCALA_VERSION-$KAFKA_VERSION.tgz -C $KAFKA_BINARIES_DIR
```
Comment (scholzj) [inline images/download_kafka.sh:25]: You need to add checksums.

File: release.version [hunk 5]
```diff
@@ -0,0 +1 @@
+0.0.1
```
Comment (scholzj) [inline release.version:1]: So do we version the images separately from the test container? Also, while in main branch, it should have some indication that it is not release but main branch build.

File: images/build_images.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        DOCKER_TAG=$PRODUCT_VERSION"-kafka-"$KAFKA_VERSION
+        echo "Building image with name: strimzi/$PROJECT_NAME:$DOCKER_TAG $KAFKA_VERSION with $SCALA_VERSION)."
+        docker build --build-arg version=$DOCKER_VERSION_ARG --build-arg KAFKA_VERSION=$KAFKA_VERSION --build-arg SCALA_VERSION=$SCALA_VERSION -t strimzi/$PROJECT_NAME:$DOCKER_TAG $DOCKERFILE_DIR
```
Comment (scholzj) [inline images/build_images.sh:33]: To speed up the build, you should maybe limit the build context to not include all the Kafka versions?

File: release.version (hunk 5 above)
Comment (see-quick) [reply release.version:1]: Hmmm yes, that's true...so do we want to hardcode value with `main`? Additionally, when we do release we basically change it to `0.0.1` or something like this...

**Pull request closed without merging**